app/uk/gov/hmrc/iossreturnsstub/controllers/EtmpController.scala

```diff
@@ -78,9 +78,11 @@ class EtmpController @Inject()(
         paymentReference = "XI/IM9001234567/2023.M11"
 
       )
-
-      Ok(Json.toJson(vatReturn)).toFuture
-
+      period match {
```

Comment:
Something like 
```scala
  def getVatReturn(iossNumber: String, period: String): Action[AnyContent] = Action.async { implicit request =>
    val validPeriods = List("23AK", "23AL")

    logger.info(s"Here's the request: ${request} ${request.headers} ${request.body}")
    jsonSchemaHelper.applySchemaHeaderValidation(request.headers) {
      validPeriods.find( _ == period).map{ validPeriod =>
        val vatReturn = EtmpVatReturn(
          returnReference = "XI/IM9001234567/2023.M11",
          periodKey = validPeriod,
          returnPeriodFrom = LocalDate.of(2023, 11, 1),
          returnPeriodTo = LocalDate.of(2023, 11, 30),
          goodsSupplied = Seq(
            EtmpVatReturnGoodsSupplied(
              msOfConsumption = "FR",
              vatRateType = EtmpVatRateType.ReducedVatRate,
              taxableAmountGBP = BigDecimal(12345.67),
              vatAmountGBP = BigDecimal(2469.13)
            )
          ),
          totalVATGoodsSuppliedGBP = BigDecimal(2469.13),
          totalVATAmountPayable = BigDecimal(2469.13),
          totalVATAmountPayableAllSpplied = BigDecimal(2469.13),
          correctionPreviousVATReturn = Seq(
            EtmpVatReturnCorrection(
              periodKey = "23AJ",
              periodFrom = LocalDate.of(2023, 10, 1).toString,
              periodTo = LocalDate.of(2023, 10, 31).toString,
              msOfConsumption = "FR"
            )
          ),
          totalVATAmountFromCorrectionGBP = BigDecimal(100.00),
          balanceOfVATDueForMS = Seq(
            EtmpVatReturnBalanceOfVatDue(
              msOfConsumption = "FR",
              totalVATDueGBP = BigDecimal(2569.13),
              totalVATEUR = BigDecimal(2569.13)
            )
          ),
          totalVATAmountDueForAllMSEUR = BigDecimal(2569.13),
          paymentReference = "XI/IM9001234567/2023.M11"

        )

        Ok(Json.toJson(vatReturn)).toFuture
      }.getOrElse{
        logger.warn(s"period $period is not a valid period {${validPeriods.mkString(", ")}}")

        NotFound(Json.toJson("")).toFuture
      }
    }
  }

```
Might be clearer? The value is only set once and doesn't meander. Valid periods are also clearer at the top

Reply:
Thanks @pbyrne84